tests/runtimes/simple-keyvalue/src/methods.rs

```diff
@@ -350,6 +356,92 @@ impl Methods {
             .transpose()
             .map_err(|err| err.to_string())
     }
+
+    /// ElGamal encryption.
+    pub fn encrypt(ctx: &mut TxContext, args: Encrypt) -> Result<Option<String>, String> {
+        if ctx.is_check_only() {
+            return Ok(None);
+        }
+
+        // Derive key pair ID based on the given ID.
+        let key_pair_id = KeyPairId::from(Hash::digest_bytes(args.key_pair_id.as_bytes()).as_ref());
+
+        // Fetch public key.
+        let io_ctx = IoContext::create_child(&ctx.parent.core.io_ctx);
+        let result =
+            ctx.parent
+                .key_manager
+                .get_public_ephemeral_key(io_ctx, key_pair_id, args.epoch);
+        let long_term_pk = tokio::runtime::Handle::current()
+            .block_on(result)
+            .map_err(|err| err.to_string())?
+            .ok_or("public ephemeral key not available")?;
+
+        // Generate ephemeral key.
+        let mut rng = OsRng {};
+        let ephemeral_sk = x25519_dalek::StaticSecret::new(&mut rng);
+        let ephemeral_pk = x25519_dalek::PublicKey::from(&ephemeral_sk);
+
+        // ElGamal encryption.
+        let ciphertext = deoxysii::box_seal(
+            &[0u8; NONCE_SIZE],
+            args.plaintext.into_bytes(),
+            vec![],
+            &long_term_pk.key.0,
+            &ephemeral_sk.to_bytes(),
+        )
+        .map_err(|err| format!("failed to encrypt plaintext: {}", err))?;
+
+        // Encode ephemeral_pk || ciphertext to Base64.
+        let mut v = ephemeral_pk.as_bytes().to_vec();
+        v.extend(ciphertext);
+        let ciphertext = base64::encode(v);
```

Comment:
Why base64 encoding? Everything is serialized into CBOR which is a binary format anyway.

Reply:
Wanted to be consistent with plaintext's type, plus it is easier to log the ciphertext if you want/need to debug. Will change to `[]byte` everywhere, so that encryption is not limited to utf8.